### PR TITLE
Feature/filestorage sharetype create

### PR DESCRIPTION
**Pull request by ehdou**
Add support for Share Type Create
For #140

### CODE REVIEW

**Comment by ehdou** [comment]
Share type [DB](https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/db/sqlalchemy/models.py#L470) representation, [view](https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/api/views/types.py) and [create function](https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/api/v2/share_types.py#L132)

**Comment by ehdou** [comment]
@jrperritt This PR is ready for review. It adds support for share list Create

File: openstack/sharedfilesystems/v2/sharetypes/requests.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+// CreateOpts contains options for creating a ShareType. This object is
+// passed to the sharetypes.Create function. For more information about
+// these parameters, see the ShareType object.
+type CreateOpts struct {
+	// The share type name
+	Name string `json:"name" required:"true"`
+	// Indicates whether a share type is publicly accessible
+	OSShareTypeAccessIsPublic bool `json:"os-share-type-access:is_public"`
```
Comment (jrperritt) [inline openstack/sharedfilesystems/v2/sharetypes/requests.go]: Let's just call this `IsPublic`.

File: openstack/sharedfilesystems/v2/sharetypes/requests.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	// The extra specifications for the share type
+	ExtraSpecs ExtraSpecsOpts `json:"extra_specs" required:"true"`
+}
+
+type ExtraSpecsOpts struct {
+	// An extra specification that defines the driver mode for share server, or storage, life cycle management
+	DriverHandlesShareServers bool `json:"driver_handles_share_servers" required:"true"`
+	// An extra specification that filters back ends by whether they do or do not support share snapshots
+	SnapshotSupport bool `json:"snapshot_support,omitempty"`
```
Comment (jrperritt) [inline openstack/sharedfilesystems/v2/sharetypes/requests.go]: If we want to have `omitempty` here, we'll probably want to make this type a `*bool`.

File: openstack/sharedfilesystems/v2/sharetypes/requests.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	Name string `json:"name" required:"true"`
+	// Indicates whether a share type is publicly accessible
+	OSShareTypeAccessIsPublic bool `json:"os-share-type-access:is_public"`
+	// The extra specifications for the share type
+	ExtraSpecs ExtraSpecsOpts `json:"extra_specs" required:"true"`
+}
+
+type ExtraSpecsOpts struct {
+	// An extra specification that defines the driver mode for share server, or storage, life cycle management
+	DriverHandlesShareServers bool `json:"driver_handles_share_servers" required:"true"`
```
Comment (jrperritt) [inline openstack/sharedfilesystems/v2/sharetypes/requests.go]: :+1:

File: openstack/sharedfilesystems/v2/sharetypes/results.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+// ShareType contains all the information associated with an OpenStack
+// ShareType.
+type ShareType struct {
+	// The Share Type ID
+	ID string `json:"id"`
+	// The Share Type name
+	Name string `json:"name"`
+	// Indicates whether a share type is publicly accessible
+	OSShareTypeAccessIsPublic bool `json:"os-share-type-access:is_public"`
```
Comment (jrperritt) [inline openstack/sharedfilesystems/v2/sharetypes/results.go]: `IsPublic`

File: openstack/sharedfilesystems/v2/sharetypes/requests.go (hunk 1 above)
Comment (ehdou) [inline openstack/sharedfilesystems/v2/sharetypes/requests.go]: done

File: openstack/sharedfilesystems/v2/sharetypes/requests.go (hunk 2 above)
Comment (ehdou) [inline openstack/sharedfilesystems/v2/sharetypes/requests.go]: done

File: openstack/sharedfilesystems/v2/sharetypes/results.go (hunk 4 above)
Comment (ehdou) [inline openstack/sharedfilesystems/v2/sharetypes/results.go]: done

**Comment by jrperritt** [comment]
+2

**Pull request merged**